### ISSUE TITLE
Some code problems

### ISSUE THREAD

**Issue by wxcnb** [decisionintelligence/CATCH#9]
I copied the CATCH model in the code you provided and wrote the training inference function myself for direct use in training and inference (your factory pattern is relatively difficult to read and modify). However, two problems were encountered in freq_rec_loss. One is that the dim parameter was not set when frequency_loss was used in CATCH.py, so an error was reported at the forward mean calculation. Secondly, during training, loss_auxi takes the mean for the specified dimension and still retains a certain shape, thus reporting an error in backpropagation. Can I directly remove the part related to the dim parameter? Because in the process of calculating the loss, this parameter seems to be useless.

**Comment by ccloud0525** [comment]
> I copied the CATCH model in the code you provided and wrote the training inference function myself for direct use in training and inference (your factory pattern is relatively difficult to read and modify). However, two problems were encountered in freq_rec_loss. One is that the dim parameter was not set when frequency_loss was used in CATCH.py, so an error was reported at the forward mean calculation. Secondly, during training, loss_auxi takes the mean for the specified dimension and still retains a certain shape, thus reporting an error in backpropagation. Can I directly remove the part related to the dim parameter? Because in the process of calculating the loss, this parameter seems to be useless.

Thank you for your attention. Of course you can do this. In fact, part of the freq_rec_loss.py is referenced from FredDF, so there may be some unnecessary parameters.
